pallets/manta-pay/src/lib.rs

```diff
@@ -953,14 +1001,27 @@ pub enum ReceiverLedgerError {
     ///
     /// The asset has already been registered with the ledger.
     AssetRegistered,
+
+    /// Type Marker Parameter
+    Marker(PhantomData<T>),
```

Comment:
PhantomData as enum field looks problematic to me, specifically if one was to iterate over all enum members (see other comment) because now `ReceiverLedgerError::Marker` is a valid enum field.
There are a few attempts at working around this [here](https://github.com/rust-lang/rust/issues/32739#issuecomment-205656212)

(the discussion on `exhaustive_patterns` in that issue features a cameo from one of our very own https://github.com/rust-lang/rust/issues/51085#issuecomment-974479661 )

Leaving this as resolved since we don't do iteration in this PR anyway